### ISSUE TITLE
Implement FAASTA

### ISSUE THREAD

**Issue by zaccharieramzi** [CEA-COSMIC/ModOpt#45]
The [FAASTA](https://arxiv.org/pdf/1512.06999.pdf) algorithm allows us to perform a FISTA algorithm in the analysis formulation. In our case we could see if the FAASTA still performs well with the [new restarting options](https://github.com/CEA-COSMIC/ModOpt/issues/34).

Since the FISTA with an orthogonal transform is way faster than the Condat-Vu, we are going to see now if this holds true even when we have to compute an extra inner iteration.

**Comment by sfarrens** [comment]
@zaccharieramzi are you planning a PR for this issue?

**Comment by zaccharieramzi** [comment]
Yes I am. I am currently working on the best way to package it but it's basically done. Do you want to have a WIP PR to see if the direction is statisfying?

**Comment by zaccharieramzi** [comment]
@sfarrens even with the trimming (#50 ), the FAASTA is still way slower than Condat for reconstruction problems with undecimated wavelets. Should we still have it in Modopt?

**Comment by sfarrens** [comment]
@zaccharieramzi I think having a diverse set of reconstruction algorithms would be useful regardless. So if you have already implemented it then feel free to open a PR.

**Comment by zaccharieramzi** [comment]
Ok I will package it in a PR.

**Comment by sfarrens** [comment]
@zaccharieramzi are you still planning a PR for this algorithm?

**Comment by zaccharieramzi** [comment]
Not at the moment. Do you still want to keep this open? I will probably do it at some point but I don't think I can find the time right now.

**Comment by sfarrens** [comment]
We can keep it open, I just wanted to check on the status. :)
